### PR TITLE
Add section about not spreading props

### LINKED ISSUE #1443: Add section about not spreading props
Spread props should be avoided, except for exceptions like:
 - HOCs that proxy down props and hoist propTypes
 - spreading objects with known, explicit props - like if you construct an object literal (statically or via a "pick" pattern) and then promptly spread it
 - others?

Let's add a section to the React guide about this.

### PR DESCRIPTION
**Pull request by maggielove**
Fixes #1443

### CODE REVIEW

**Comment by maggielove** [comment]
cc @ljharb

File: react/README.md [hunk 1]
```diff
@@ -380,6 +380,58 @@
   };
   ```
 
+  - Use spread props sparingly.
+  > Why? Otherwise you're more likely to pass unnecessary props down to components. And for React 15 and older, you could [pass invalid HTML attributes to the DOM](https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html).
```
Comment (ljharb) [inline react/README.md]: I think it's React < 15.2 or 15.3? Let's be specific here :-)

**Comment by maggielove** [comment]
Thanks @ljharb ! This is ready for re-review. :) I'm not sure why the Travis build is failing--looking into it.

**Comment by ljharb** [comment]
Tests will now pass once this is rebased on master.

**Comment by maggielove** [comment]
@ljharb Whenever you get a chance, I made your suggested changes. What do you think -- does it need more edits?

**Pull request merged**